`double` specifier should be allowed to combine with `long`

We allow `long double` but not `double long`:

```c
double long x;
```

```
test.c:1:8: error: cannot combine with previous 'double' specifier
double long x;
       ^
```

--- a/src/aro/Type.zig
+++ b/src/aro/Type.zig
@@ -2091,6 +2091,7 @@ pub const Builder = struct {
             },
             .long => b.specifier = switch (b.specifier) {
                 .none => .long,
+                .double => .long_double,
                 .long => .long_long,
                 .unsigned => .ulong,
                 .signed => .long,
@@ -2103,6 +2104,7 @@ pub const Builder = struct {
                 .complex_long => .complex_long_long,
                 .complex_slong => .complex_slong_long,
                 .complex_ulong => .complex_ulong_long,
+                .complex_double => .complex_long_double,
                 else => return b.cannotCombine(p, source_tok),
             },
             .int128 => b.specifier = switch (b.specifier) {

--- a/test/cases/double long.c
+++ b/test/cases/double long.c
@@ -1,0 +1,2 @@
+_Complex double long x;
+double long y;

